docs/utilities/config_annotations.md

```diff
@@ -0,0 +1,234 @@
+
+!! important
+
+    By default the annotation-based configuration system loads all mod configurations exactly once, directly after the mod is constructed but before the mod enters pre-initialization.
+
+
+Comments
+-------------------------
+
+Often, despite our best efforts in naming, the property key alone does not do justice in describing the function of a given configuration property. In this case, or moreso in general it would be nice as to be able to give the end-user more information as to what a configuration element actually does. For this reason, the `@Comment` annotation was created.
+
+The `@Comment` annotation is simple to use, as it may be applied directly to any valid field inside a `@Config` class as to add either a single line, or a multi-line comment. Extending the prior example usage of the `@Config` annotation:
+
+````Java
+@Config(modid = "my_mod_id")
+public class MyModConfig {
+
+  @Comment("This is a lovely single-line comment.")
+  public static boolean myPropertyWithSingleLineComment = false;
+
+  @Comment({"Whereas this,", "is a multi-line comment"})
+  public static boolean myPropertyWithMultiLineComment = false;
+
+}
+````
+
+!!! note
+
+    Single-line comments, having only one string value set in the annotation, do not require (but will accept) surrounding brackets unlike multi-line comments that require surrounding brackets.
+
+!!! important
+
+    It is strongly recommended that you do not use the newline character `\n` (or platform-specific equivalents) in comments, and instead use the multi-line `@Comment` annotation form, as Forge will then be able to ensure the proper handling of the multi-line comment.
+
+Property Ranges
+-------------------------
+
+While when using boolean configuration properties all possible input values are considered valid, in the case of numeric inputs there may arise the need as to constrain the acceptable inputs to a certain continuous range. For this reason, the annotations `@RangeInt` and `@RangeDouble` exist.
+
+The `@Range` annotations consist of two values, a minimum value and a maximum value, which together define an inclusive range of acceptable inputs for a given numeric property. Note, that inclusive means that the values set for `min` and `max` are considered valid inputs for the property. If either min or max are not set, then the annotation will default to `<? extends Number>.MIN_VALUE` and `<? extends Number>.MAX_VALUE` respectively.
+
+Building off the prior examples a sample usage of the `@RangeInt` and `@RangeDouble` would be as follows.
+
+````Java
+@Config(modid = "my_mod_id")
+public class MyModConfig {
+
+  @Comment("This integer property accepts values 1 to 10 inclusive.")
+  @RangeInt(min = 1, max = 10)
+  public static int myIntegerProperty = false;
+
+  @Comment("This double property accepts values 0.0 to 1.0 inclusive.")
+  @RangeDouble(min = 0.0, max = 1.0)
+  public static double myDoubleProperty = false;
+
```

Comment:
I think these code examples are getting a bit superfluous at this point, try to keep the docs more focused on documenting, not examples.

Reply:
Or at the least remove the unnecessary bits like the class definition and `@Comment`

Reply:
The goal was consistency within each of the sections, consisting of a description followed by a code example finalized by notes (i.e. important, note).